### PR TITLE
Feat/homepage

### PR DESCRIPTION
**Pull request by Malik705017**
# Description

Complete homepage wireframe

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Open http://localhost:3000/
截圖 2021-10-21 下午2 22 08

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by Malik705017** [comment]
@Nick0603 @AlanSyue 再麻煩你們幫我 review，thx!

File: styles/Home.module.scss [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+  
+  .feature {
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    align-items: center;
+
+    width: 300px;
+    max-width: 20%;
+    aspect-ratio: 1 / 1;
```
Comment (AlanSyue) [inline styles/Home.module.scss:56]: [curiosity]
Do `1 / 1` need to be declared?

**Review by AlanSyue — APPROVED** [review]

File: components/atoms/Card/index.module.scss [hunk 2]
```diff
@@ -0,0 +1,38 @@
+@import "../../../styles/variables.scss";
```
Comment (Nick0603) [inline components/atoms/Card/index.module.scss:1]: Structure Discussion:

The file naming of `index.tsx` and `index.module.scss` are not easy to know the file content directly.

For example, if I search some keyword and result is `all index files`
image

The solution is like `hello-world` component. We define the all files are starting from `hello-world`
image

And export in index file, and we can easy to know what are exported.
image

File: components/molecules/BannerContainer/index.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+          <p>報名時間：2022/6/1-7/1</p>
+          <Button text="立即報名" className={styles.button} reverse />
+        </div>
+        <div className={styles.summaryInfo}></div>
+      </div>
+    </>
+  );
+};
+
+export default BannerContainer;
```
Comment (Nick0603) [inline components/molecules/BannerContainer/index.tsx:29]: The `BannerContainer` can implemented by `SectionWrapper`.

File: styles/mixin.scss [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  @media(max-width:1023px){
+    @content;
+  }
+}
+
+@mixin mobile{
+  @media(max-width:767px){
+      @content;
+  }
+}
```
Comment (Nick0603) [inline styles/mixin.scss:23]: Make easier to read.
Can write the variables to global variables.scss
The logic of `desktop` here looks like a `laptop` logic. 


```suggestion
$width-desktop: 1960px;
$width-laptop: 1440px;
$width-large-tablet: 1024px;
$width-tablet: 768px;

@mixin laptop{
  @media(max-width: ($width-desktop - 1px)){
    @content;
  }
}

@mixin large-tablet{
  @media(max-width: ($width-laptop - 1px)){
    @content;
  }
}

@mixin tablet{
  @media(max-width: ($width-large-tablet - 1px)){
    @content;
  }
}

@mixin mobile{
  @media(max-width: ($width-tablet - 1px)){
      @content;
  }
}

```

File: components/molecules/BannerContainer/index.tsx [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+import Banner from "../../atoms/Banner";
+import Button from "../../atoms/Button";
+import { SectionBigTitle, SubTitle } from "../../atoms/SectionTitle";
+
+import styles from "./index.module.scss";
+
+const BannerContainer: FC = () => {
+  return (
+    <>
```
Comment (Nick0603) [inline components/molecules/BannerContainer/index.tsx:11]: Redundant code

**Review by Nick0603 — CHANGES_REQUESTED** [review]
Maybe we need to have a meeting to discuss.

File: components/atoms/CollapseBox/CollapseBox.tsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+  children,
+}) => {
+  const [isOpen, setOpen] = useState(false);
+
+  return (
+    <div className={classnames(styles.collapseBox, className)}>
+      <div
+        className={classnames(
+          styles.collapseHeader,
+          isOpen ? styles.open : styles.close
```
Comment (Nick0603) [inline components/atoms/CollapseBox/CollapseBox.tsx:27]: Seem no `close` key in styles?

File: components/atoms/CollapseBox/CollapseBox.tsx [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+        )}
+        onClick={() => setOpen((isOpen) => !isOpen)}
+      >
+        <span className={styles.title}>{title}</span>
+        <Image
+          alt="arrow"
+          src={isOpen ? arrowUp : arrowDown}
```
Comment (Nick0603) [inline components/atoms/CollapseBox/CollapseBox.tsx:34]: [minor]
Would you like to make a rotation animation for it?
If you want it, you can use same image and set `transform: rotate(180deg)`

File: pages/index.tsx [hunk 8]
```diff
@@ -1,71 +1,110 @@
...
-            <p>Learn about Next.js in an interactive course with quizzes!</p>
-          </a>
-
-          <a
-            href="https://github.com/vercel/next.js/tree/master/examples"
-            className={styles.card}
+const Home: NextPage = () => {
+  return (
+    <>
+      <BannerContainer backgroundImage="images/Banner-Homepage.png" />
```
Comment (Nick0603) [inline pages/index.tsx:17]: The image naming is weird. How do we use `snake case` for it?

File: pages/index.tsx [hunk 9]
```diff
@@ -1,71 +1,110 @@
...
-        <p className={styles.description}>
-          Get started by editing{" "}
-          <code className={styles.code}>pages/index.js</code>
-        </p>
+import Button from "../components/atoms/Button/Button";
+import { SubTitle } from "../components/atoms/SectionTitle/SectionTitle";
+import Card from "../components/atoms/Card/Card";
+import CollapseBox from "../components/atoms/CollapseBox/CollapseBox";
+import SectionWrapper from "../components/molecules/SectionWrapper/SectionWrapper";
+import BannerContainer from "../components/molecules/BannerContainer/BannerContainer";
```
Comment (Nick0603) [inline pages/index.tsx:10]: Add `index.tsx` for each component to make clear for exporter and make easier for imoprter.

```suggestion
import Button from "../components/atoms/Button";
import { SubTitle } from "../components/atoms/SectionTitle";
import Card from "../components/atoms/Card";
import CollapseBox from "../components/atoms/CollapseBox";
import SectionWrapper from "../components/molecules/SectionWrapper";
import BannerContainer from "../components/molecules/BannerContainer";
```

**Review by Nick0603 — COMMENTED** [review]
Some suggestions

**Review by Nick0603 — APPROVED** [review]
LGTM

**Pull request merged**